config/config.py

```diff
@@ -621,10 +622,36 @@ def cublas_library():
     Does not actually run the resulting exe, to allow compiling with CUDA on a
     machine without GPUs.
     '''
-    libs = '-lcublas -lcudart'
+    # Find CUDA to add -I, -L, -rpath flags.
+    # CUDA_PATH is used in NVIDIA Getting Started documentation;
+    # CUDA_HOME is used in Spack CUDA package;
+    # else infer from `which nvcc`.
+    cuda_path = environ['CUDA_PATH']
+    if (not cuda_path):
+        cuda_path = environ['CUDA_HOME']
+    if (not cuda_path):
+        nvcc_path = shutil.which( 'nvcc' )
+        if (nvcc_path):
+            (bin_path, nvcc) = os.path.split( nvcc_path )
+            (cuda_path, bin_) = os.path.split( bin_path )
+
+    cxxflags = define('HAVE_CUBLAS')
+    ldflags  = ''
+    libs     = '-lcublas -lcudart'
+
+    if (cuda_path):
+        incdir = os.path.join( cuda_path, 'include' )
+        cxxflags += ' -I' + incdir
+
+        libdir = os.path.join( cuda_path, 'lib64' )
+        if (not os.path.exists( libdir )):
+            libdir = os.path.join( cuda_path, 'lib' )
```

Comment:
I know I am pedantic but why do not we check is `lib` exist as well?

Reply:
Something like this (also checking incdir)?
```
        if (os.path.exists( incdir )):
            cxxflags += ' -I' + incdir

        libdir = os.path.join( cuda_path, 'lib64' )
        if (not os.path.exists( libdir )):
            libdir = os.path.join( cuda_path, 'lib' )
            if (not os.path.exists( libdir )):
                libdir = None
        if (libdir):
            ldflags += '-L' + libdir + ' -Wl,-rpath,' + libdir
```

Reply:
Or this should work, it just checks lib64 twice if it exists:
```
        libdir = os.path.join( cuda_path, 'lib64' )
        if (not os.path.exists( libdir )):
            libdir = os.path.join( cuda_path, 'lib' )
        if (os.path.exists( libdir )):
            ldflags += '-L' + libdir + ' -Wl,-rpath,' + libdir
```

Reply:
I updated to the 2nd choice.